python/ray/data/tests/test_dataset.py

```diff
@@ -391,6 +391,51 @@ def test_batch_tensors(ray_start_regular_shared):
     assert df.to_dict().keys() == {"value"}
 
 
+def test_arrow_block_select():
```

Comment:
Existing `test_groupby_xxx()` tests in this file already gave a good coverage of the added change (actually I fixed bunch of bugs by running these tests), so didn't add a new one for groupby here.